gpii/node_modules/packagekit/nodepackagekit/nodepackagekit.cc

```diff
@@ -273,25 +294,31 @@ Handle<Value> performAction(const Arguments& args) {
                                    (PkProgressCallback) progressCallback,
                                    NULL, &err);
   } else {
-      ThrowException(Exception::Error(String::New(
-          "You have to provide the action to be performed, either 'install'," 
-          "'update' or 'remove'")));
+      Nan::ThrowError(
+          "You have to provide the action to be performed, either 'install', " 
+          "'update' or 'remove'");
   }
```

Comment:
@javihernandez There is a potential memory leak here.  The `task` and `package_ids` were allocated above, and are properly freed just below this throw block.  However, if the throw is executed, they are not freed.

Patch coming up.